src/comparator/src/ArtifactDelta.ts

```diff
@@ -57,6 +57,14 @@ export default class ArtifactDelta {
     return this._hashChanged;
   }
 
+  public get hashChangeUnexpected(): boolean {
+    return (
+      this._hashChanged &&
+      this.failingBudgets.length === 0 &&
```

Comment:
I'm curious why we need to check for this (no failing budgets). Surely budgets are an orthogonal concern? Whether or not we have a failing budget, the hash change may have changed unexpectedly.

Reply:
This was pulled from previous logic in which we prefer a budget failure over a hash change. I think you're right that it's an orthogonal concern and could be pulled out — just need to make sure that everywhere it's displayed, the budget failures are shown as a priority